federatedscope/core/auxiliaries/data_builder.py

```diff
@@ -611,3 +611,17 @@ def merge_data(all_data):
                 merged_data[d_name][elem_name])
 
     return merged_data
+
+
+def merge_test_data(all_data):
+
+    merged_data = dict()
+    merged_data['test'] = all_data[1]['test']
+    num_of_sample_per_client = [len(all_data[1]['test'].dataset)]
```

Comment:
@rayrayraykk Is this general enough? e.g., for node classification dataset

Reply:
Since global eval is True in node/link tasks, we do not need set `federate.merge_test_data=True`. But if global eval is False in node/link tasks, `federate.merge_test_data=True` might raise errors.

Reply:
IMO, the format of the test set in graph federated learning is quiet different from others

Reply:
If node/link-level tasks are exceptional, we'd better exclude them in assertions about `merge_test_data`.

Reply:
I have reused `merge_data` in this [pr](https://github.com/alibaba/FederatedScope/pull/220), which would raise NotImplementedError for node/link-level tasks